aixcalibuha/sensitivity_analysis/sobol.py

```diff
@@ -68,13 +85,301 @@ def generate_samples(self):
                             N=self.num_samples,
                             **self.create_sampler_demand())
 
-    def _get_res_dict(self, result: dict, cal_class: CalibrationClass):
+    def _save(self, result, time_dependent=False):
+        if time_dependent:
+            savepath_result_1 = self.cd.joinpath(f'{self.__class__.__name__}_results_time.csv')
+            savepath_result_2 = self.cd.joinpath(f'{self.__class__.__name__}_results_second_order_time.csv')
+        else:
+            savepath_result_1 = self.cd.joinpath(f'{self.__class__.__name__}_results.csv')
+            savepath_result_2 = self.cd.joinpath(f'{self.__class__.__name__}_results_second_order.csv')
+        if not result[0].empty:
+            result[0].to_csv(savepath_result_1)
+            self.reproduction_files.append(savepath_result_1)
+        if not result[1].empty:
+            result[1].to_csv(savepath_result_2)
+            self.reproduction_files.append(savepath_result_2)
+
+    def _conv_local_results(self, results: list, local_classes: list, verbose=False):
+        _conv_results = []
```

Comment:
Again, would super be an option?
Also, can you add a small docstring to _save and _conv_local_results in both sobol and the base sa?

Reply:
Here, to use super is not directly possibly, and I would have to restructure some other parts. So, I would leave it as it is because I don't know how much work it is for this small improvement. But I added the missing docstrings.